FreeRTOS/FreeRTOS-Kernel
# Let each stream/message can use its own sbSEND_COMPLETED

Description
-----------
According to issue #312, this PR provides `configUSE_SB_COMPLETED_CALLBACK` that makes each stream/message buffer use its own `sbSEND_COMPLETED`

Test Steps
-----------
In `FreeRTOSconfig.h`,
If `configUSE_SB_COMPLETED_CALLBACK` is not set or set to 0, the single global definition of `sbSEND_COMPLETED` applies to every stream/message buffer.
If `configUSE_SB_COMPLETED_CALLBACK` is set to 1, each stream/mesasage buffer can define a callback function as its own `sbSEND_COMPLETED`, Users can pass the callback function as the argument(if pass NULL, it will use the default implementaion) when the buffer is (static)created:
```
void func1()
{
  /* User-defined sbSEND_SOMPLETED */
}
xStreamBuffer = xStreamBufferCreate( xBufferSizeBytes, xTriggerLevelBytes, func1)
```
The default type of the callback fucntion is "void function()", users can change the type of the callback function in `stream_buffer.h`:
```
#if (configUSE_SB_COMPLETED_CALLBACK == 1)
    typedef void (* StreamBufferCallbackFunction_t)();   //Declare a function pointer that points to the stream buffer's callback
#endif
```
* Validated with STM32F407VG

Related Issue
-----------
Issue #312


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
